### ISSUE TITLE
🐛 [SambaNAS 10.0.2-nas4 ] <major issue with media library >

### ISSUE THREAD

**Issue by RicFried** [dianlight/hassio-addons#178]
### Addon

SambaNAS

### Description

Add on installed OK
Protection mode switch set to OFF

2, even 3 USB drives mounted OK

medialibrary is set to enable : true
drives show up in library OK

Issue manifests after a REBOOT of Home Assistant.

Addon starts. 
in the log this error shows up
ERROR: Got unexpected response from the API: A mount already exists with name 5TB_HD
then the Add on gracefully stops.

Upon further troubleshooting, I see the drives (folders) in the MEDIA folder
attempting to delete these from media folder not possible

via SSH tried the following:
umount -f 5TB_USB
umount -f BACKUP_MAC
rm -r 5TB_USB
rm -r BACKUP_MAC

restarted the addon but same error.
rebooted HASS and these folders (or mount points shows up again)

apparently after a reboot the medialibrary feature prevents the addon to start.

by setting medialibrary is set to enable : false
the addon starts.

we need the library on, and so, ability to clean up these mount points when restarting the add-on, prior to re-adding them.

(i use a dedicated raspberry PI4-B with home-assistant OS latest version)

### Reproduction steps

```bash
1. start add on with media library set to true
2. verify usb drives are mounted
3. reboot hass host
4. error in the log 
   ERROR: Got unexpected response from the API: A mount already exists with name ____
5. addon will stop and cannot be used anymore with library set to true / enabled
```

### Addon Logs

```bash
[22:12:17] INFO: Enabling S.M.A.R.T for /dev/sdb
smartctl 7.3 2022-02-28 r5338 [aarch64-linux-6.1.21-v8] (local build)
Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org

tdbsam_open: Converting version 0.0 database to version 4.0.
tdbsam_convert_backup: updated /var/lib/samba/private/passdb.tdb file.
Added user Ric.
=== START OF ENABLE/DISABLE COMMANDS SECTION ===
SMART Enabled.
SMART Attribute Autosave Enabled.
---------------------------------------------------
[22:12:18] INFO: Exposed Disks Summary:
[config]                path = /config # TM:false USAGE: #
[addons]                path = /addons # TM:false USAGE: #
[ssl]                   path = /ssl # TM:false USAGE: #
[share]                 path = /share # TM:false USAGE: #
[backup]                path = /backup # TM:false USAGE: #
[media]                 path = /media # TM:false USAGE: #
[5TB_USB]               path = /mnt/5TB_USB # TM:true USAGE:media #
[BACKUP_MAC]            path = /mnt/BACKUP-MAC # TM:true USAGE:media #
---------------------------------------------------
s6-rc: info: service init-samba successfully started
s6-rc: info: service init-mqtt: starting
s6-rc: info: service wsdd: starting
s6-rc: info: service smbd: starting
s6-rc: info: service nmbd: starting
s6-rc: info: service avahi: starting
s6-rc: info: service wsdd successfully started
s6-rc: info: service smbd successfully started
s6-rc: info: service avahi successfully started
s6-rc: info: service nmbd successfully started
s6-rc: info: service cifs-supervisor-mount: starting
[22:12:18] INFO: Automount Shares...
[22:12:18] INFO: Starting the AVAHI for homeassistant...
Waiting for daemon ...
SMART Automatic Offline Testing Enabled every four hours.
[22:12:19] INFO: MQTT support not enabled in config
s6-rc: info: service init-mqtt successfully started
[22:12:19] INFO: Starting the wsdd daemon for WORKGROUP/homeassistant...

s6-rc: info: service init-smartd successfully started
[22:12:19] INFO: Interfaces: -i end0 -i wlan0 
Wait Samba Server to going up..(max 30s)
2023-11-08 22:12:20,043:wsdd INFO(pid 356): using pre-defined UUID cb6a33a0-b841-531d-a34e-09ecec492eb1
2023-11-08 22:12:20,047:wsdd INFO(pid 356): joined multicast group 239.255.255.250 on 192.168.100.25%end0
2023-11-08 22:12:20,054:wsdd INFO(pid 356): scheduling Hello message via end0 to 239.255.255.250%end0
2023-11-08 22:12:20,137:wsdd INFO(pid 356): joined multicast group [ff02::c] on fe80::ab67:748:dd2:64c6%end0
2023-11-08 22:12:20,140:wsdd INFO(pid 356): scheduling Hello message via end0 to ff02::c%end0
nmbd version 4.18.4 started.
Copyright Andrew Tridgell and the Samba Team 1992-2023
smbd version 4.18.4 started.
Copyright Andrew Tridgell and the Samba Team 1992-2023
INFO: Profiling support unavailable in this build.
[22:12:24] ERROR: Got unexpected response from the API: A mount already exists with name 5TB_USB
s6-rc: warning: unable to start service cifs-supervisor-mount: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
/run/s6/basedir/scripts/rc.init: fatal: stopping the container.
s6-rc: info: service init-smartd: stopping
s6-rc: info: service init-mqtt: stopping
s6-rc: info: service wsdd: stopping
s6-rc: info: service smbd: stopping
s6-rc: info: service nmbd: stopping
s6-rc: info: service hd-idle: stopping
2023-11-08 22:12:24,127:wsdd INFO(pid 356): received termination/interrupt signal, tearing down
2023-11-08 22:12:24,127:wsdd INFO(pid 356): shutting down gracefully...
2023-11-08 22:12:24,129:wsdd INFO(pid 356): scheduling Bye message via end0 to 239.255.255.250%end0
s6-rc: info: service avahi: stopping
Got SIGTERM: going down...
s6-rc: info: service init-smartd successfully stopped
Got SIGTERM, quitting.
s6-rc: info: service init-mqtt successfully stopped
[22:12:24] INFO: smbd stopped
[22:12:24] INFO: hd-idle stopped
s6-rc: info: service smbd successfully stopped
[22:12:24] INFO: avahi stopped
s6-rc: info: service hd-idle successfully stopped
[22:12:24] INFO: nmbd stopped
s6-rc: info: service avahi successfully stopped
s6-rc: info: service nmbd successfully stopped
2023-11-08 22:12:24,995:wsdd INFO(pid 356): Done.
[22:12:25] INFO: wsdd stopped
s6-rc: info: service wsdd successfully stopped
s6-rc: info: service init-samba: stopping
s6-rc: info: service init-samba successfully stopped
s6-rc: info: service init-automount: stopping
[22:12:26] INFO: Unmount drivers:
/mnt/5TB_USB
/mnt/BACKUP-MAC
[22:12:26] INFO: Bye.
s6-rc: info: service init-automount successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service base-addon-log-level: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service base-addon-log-level successfully stopped
s6-rc: info: service base-addon-banner: stopping
s6-rc: info: service base-addon-banner successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stoppedUSB
```

### Addon Config

```bash
workgroup: WORKGROUP
username: Ric
allow_hosts:
  - 10.0.0.0/8
  - 172.16.0.0/12
  - 192.168.0.0/16
  - 169.254.0.0/16
  - fe80::/10
  - fc00::/7
automount: true
moredisks: []
available_disks_log: true
medialibrary:
  enable: true
veto_files:
  - ._*
  - .DS_Store
  - Thumbs.db
  - icon?
  - .Trashes
compatibility_mode: false
wsdd2: false
mqtt_use_legacy_entities: false
autodiscovery: {}
other_users: []
acl: []
interfaces: []
password: -------
```

### Architecture

armv7

### OS

HaOS

**Comment by Ivshin** [comment]
same problem

**Comment by RicFried** [comment]
I found out a "manual fix"
1. stop the add-on (if it was running)
2. go to homeassistant settings / system / storage
3. remove the drives that are showing in the network storage section
4. you can now start the add-on with media library set to true

either it's something new in homeassistant new updates, that conflicts with samba-as

a possible solution would be to add in the addon code a check at start, against these entries, and remove theses, then start the add-on

this way we don't have to do all this at every reboot.

thanks to dianlight  / Lucio Tarantino for this marvelous add-on !

**Comment by dianlight** [comment]
I hope is fix in the release 12.1.0-nas
